### ISSUE TITLE
Undocumented USER env variable and permission issues

### ISSUE THREAD

**Issue by eltiel** [jokob-sk/Pi.Alert#78]
When I tried starting up the image, the log shows:

`We need USER to be set!`

I've already set HOST_USER_ID and HOST_USER_GID

------------------

Secondly, I think I'm having permission issues despite having set the db folder to 777. Following the guide, I can't `chown root:www-data` the folder as I'm on Synology and there's no www-data group.

**Comment by jokob-sk** [comment]
Hi there

Thanks for using my image :) 

Did you follow the instructions outlined in the readme? https://github.com/jokob-sk/Pi.Alert/tree/main/dockerfiles

The image should run without issues even if you omit the "HOST_USER_ID" and "HOST_GROUP_ID" variables, they are only used for mapping the permissions for the DB. You can alternatively use the CSV export for backup. 

Are you using docker-compose for setting up the container or a different approach? If a different approach, please post yous command below (replace sensitive info). 

I'm running the container as we speak with the following docker-compose file:

```yaml
version: "3"
services:
  pialert:
    container_name: pialert
    image: "jokobsk/pi.alert:latest"      
    network_mode: "host"        
    restart: always
    volumes:
      - my/path/pialert/config:/home/pi/pialert/config
      - my/path2/pialert/db/pialert.db:/home/pi/pialert/db/pialert.db
    environment:
      - TZ=Europe/Berlin
      - PORT=20211
      - HOST_USER_ID=1055
      - HOST_USER_GID=1000
```

I just run `docker-compose up`

The `We need USER to be set!` error comes form this file: https://github.com/jokob-sk/Pi.Alert/blob/3d5895265433978f49f8e054178df4d1685c0b53/dockerfiles/user-mapping.sh#L4

Which in turn relies on the variable set in the Dockerfile itself:

https://github.com/jokob-sk/Pi.Alert/blob/3d5895265433978f49f8e054178df4d1685c0b53/Dockerfile#L4

Did you modify the Dockerfile by any chance? If yes, can you please try to use the image from DockerHub instead to verify it's working? https://registry.hub.docker.com/r/jokobsk/pi.alert

>Secondly, I think I'm having permission issues despite having set the db folder to 777. Following the guide, I can't chown root:www-data the folder as I'm on Synology and there's no www-data group.

These instructions only apply for bare-metal install, which is not supported in this fork. Only the docker-image based approach is as outlined in the readme here:  https://github.com/jokob-sk/Pi.Alert#installation

> [...] Following the guide, [...]

Can you please link which guide you are following? You still should be able to SSH into the container to perform a CHOWN inside of it:

`sudo docker exec -it pialert /bin/bash` would open the terminal inside of the container.

**Comment by eltiel** [comment]
Thanks for the very quick reply @jokob-sk.

I can confirm that I read the instructions as per https://github.com/jokob-sk/Pi.Alert/tree/main/dockerfiles. I have:

1. Downloaded pialert.conf and version.conf from the given [link](https://github.com/jokob-sk/Pi.Alert/tree/main/config).
2. In `pialert.conf` specified my network adapter and network filter.
3. Set `TZ` environment
4. Mapped `home/pi/pialert.db` and `/home/pi/pialert/config` and `/home/pi/pialert/log` to persistent paths
5. Based on your explanation above, I removed `HOST_USER_ID` and `HOST_USER_GID` so I'm no longer getting the `We need USER to be set!` error
6. Specified `PORT` env variable

I used Synology Docker GUI instead of Docker Compose. The steps involved:
1. Downloading `jokobsk/pi.alert:latest` image via the Registry (DockerHub)
2. Launch (create) a Container using the downloaded image
3. Via the GUI, did items 3, 4, 5 and 6 as per above.
4. Started the Container

I did not modify the Dockerfile, in fact I've repeatedly downloaded a fresh copy as I kept trying to make it work. The version I downloaded is from DockerHub. The guide I followed was from within Pi.Alert's Help/FAQ section:

> I get the message that the database is read only.
> Check in the Pi.Alert directory if the database folder (db) has been assigned the correct permissions:
> drwxrwx--- 2 (your username) www-data
> If the permission is not correct, you can set it again with the following commands in the terminal or the console:
> sudo chgrp -R www-data ~/pialert/db
> chmod -R 770 ~/pialert/db
> If the database is still read-only, try reinstalling or restoring a database backup from the maintenance page.

To clarify, I did not get the message that the database is read only but I did get the AJAX error so I'm guessing it's related to DB permissions based on the fact that:

1. My logs show the following:

> Traceback (most recent call last):
>   File "/home/pi/pialert/back/pialert.py", line 1659, in <module>
>     sys.exit(main())       
>   File "/home/pi/pialert/back/pialert.py", line 79, in main
>     upgradeDB()
>   File "/home/pi/pialert/back/pialert.py", line 1577, in upgradeDB
>     sql.execute("""      
> sqlite3.OperationalError: no such table: Devices

2. This [comment](https://www.reddit.com/r/selfhosted/comments/rnc0g1/comment/ic6k2sa/?utm_source=share&utm_medium=web2x&context=3) in reddit.

**Comment by jokob-sk** [comment]
Hey!

Thanks a lot for providing the details. This clears things up :) I was starting to question my already limited Docker knowledge :)

So, the here are my comments:

> Mapped home/pi/pialert/db and /home/pi/pialert/config and /home/pi/pialert/log to persistent paths

The docker container doesn't support mapping the whole `/db` folder. I didn't figure out (yet) why the permissions are messed up when you do that. However you can map the 1pialert.db` file directly. See info in the readme:

---
5. Database backup
   * The DB is stored under `/home/pi/pialert/db/pialert.db`. Map this `file` to a persistent location (see [Examples](https://github.com/jokob-sk/Pi.Alert/tree/main/dockerfiles#page_facing_up-examples) for details). If facing issues (AJAX errors, can't write to DB, etc, make sure permissions are set correctly, alternatively check the logs under `/home/pi/pialert/log`)
---

Notice how that is done in Example 1 in the [dockerfile readme](https://github.com/jokob-sk/Pi.Alert/tree/main/dockerfiles#example-1):

```yaml
...
volumes:
   ...
   - ${APP_DATA_LOCATION}/pialert/db/pialert.db:/home/pi/pialert/db/pialert.db
   ...
 ``` 

If you comment out your `home/pi/pialert/db` folder mapping, the container should start (alternatively map the file directly). You can always backup your DB via the Maintenance section (CSV and DB backup) (test a restore process as well).

> I did not modify the Dockerfile, in fact I've repeatedly downloaded a fresh copy as I kept trying to make it work. The version I downloaded is from DockerHub. The guide I followed was from within Pi.Alert's Help/FAQ section:

I will have to explicitly mention this is only applicable for the bare-metal install, not the Docker version, will fix it hopefully soon.

I hope this helps - the DB backup situation could be better, but it's a legacy code-base so making it run in Docker can be a bit tricky. 

Still, thanks for pointing out these things, will update the FAQ section and think about how to improve the setup further.

**Comment by eltiel** [comment]
> The docker container doesn't support mapping the whole /db folder. I didn't figure out (yet) why the permissions are messed up when you do that. However you can map the pialert.db file directly. See info in the readme:

I somehow managed to get it working without mapping the pialert.db file directly. I downloaded `pialert.db` from [here](https://github.com/jokob-sk/Pi.Alert/blob/main/db/pialert.db) and stored it in the `db` folder. Then via Maintenance, I backed up the DB and then restored it. Not sure why that fixes the problem, but hey it works. 

Nonetheless, thank you for this nice piece of work!
